audiocraft/data/audio_dataset.py

```diff
@@ -440,24 +443,31 @@ def __getitem__(self, index: int) -> tp.Union[torch.Tensor, tp.Tuple[torch.Tenso
                 try:
                     out, sr = audio_read(file_meta.path, seek_time, self.segment_duration, pad=False)
                     out = convert_audio(out, sr, self.sample_rate, self.channels)
-                    n_frames = out.shape[-1]
-                    target_frames = int(self.segment_duration * self.sample_rate)
-                    if self.pad:
-                        out = F.pad(out, (0, target_frames - n_frames))
-                    segment_info = SegmentInfo(file_meta, seek_time, n_frames=n_frames, total_frames=target_frames,
-                                               sample_rate=self.sample_rate, channels=out.shape[0])
                 except Exception as exc:
                     logger.warning("Error opening file %s: %r", file_meta.path, exc)
                     if retry == self.max_read_retry - 1:
                         raise
-                else:
-                    break
+                    continue
 
-        if self.return_info:
-            # Returns the wav and additional information on the wave segment
-            return out, segment_info
-        else:
-            return out
+                energy = torch.sqrt(torch.mean(out ** 2))
+                if energy < self.energy_threshold:
```

Comment:
where is `energy_threshold` defined? did you mean `self.energy_filter`?

Reply:
Yes. Of course. Good catch.